model/riscv_fdext_regs.sail

```diff
@@ -15,55 +15,30 @@
 
 /* **************************************************************** */
 /* NaN boxing/unboxing.                                             */
-/* When 16-bit floats (half-precision) are stored in 32/64-bit regs  */
-/* they must be 'NaN boxed'.                                         */
-/* When 16-bit floats (half-precision) are read from 32/64-bit regs  */
-/* they must be 'NaN unboxed'.                                       */
-/* When 32-bit floats (single-precision) are stored in 64-bit regs  */
-/* they must be 'NaN boxed' (upper 32b all ones).                   */
-/* When 32-bit floats (single-precision) are read from 64-bit regs  */
-/* they must be 'NaN unboxed'.                                      */
-
-function canonical_NaN_H() -> bits(16) = 0x_7e00
-function canonical_NaN_S() -> bits(32) = 0x_7fc0_0000
-function canonical_NaN_D() -> bits(64) = 0x_7ff8_0000_0000_0000
```

Comment:
//                                      sign  exponent    significand
function canonical_NaN_H() -> bits(16) = 0b0 @ ones(5)  @ 0b1 @ zeros(9)
function canonical_NaN_S() -> bits(32) = 0b0 @ ones(8)  @ 0b1 @ zeros(22)
function canonical_NaN_D() -> bits(64) = 0b0 @ ones(11) @ 0b1 @ zeros(51)
function canonical_NaN_Q() -> bits(64) = 0b0 @ ones(15) @ 0b1 @ zeros(111)
```

Could do this instead of the match lower down?

Reply:
Good idea. I did this.